hugo/content/guides/formatting.md

```diff
@@ -0,0 +1,103 @@
+---
+title: "Formatting"
+weight: 300
+---
+
+Langium's formatting API allows to easily create formatters for your language.
+We start building a custom formatter for our language by creating a new class that inherits from `AbstractFormatter`.
+
+```ts
+import { AbstractFormatter, AstNode, Formatting } from 'langium';
+
+export class CustomModelFormatter extends AbstractFormatter {
+    protected format(node: AstNode): void {
+        // This method is called for every AstNode in a document
+    }
+}
+...
+// Bind the class in your module
+export const CustomModelModule: Module<CustomModelServices, PartialLangiumServices> = {
+    lsp: {
+        Formatter: () => new CustomModelFormatter()
+    }
+};
+```
+
+The entry point for the formatter is the abstract `format(AstNode)` method. The `AbstractFormatter` calls this method for every node of our model.
+To perform custom formatting for every type of node, we will use pattern matching.
+In the following example, we will take a closer look at a formatter for the [domain-model](https://github.com/langium/langium/tree/main/examples/domainmodel) language.
+In particular, we will see how we can format the root of our model (`DomainModel`) and each nested element (`Entity` and `PackageDeclaration`).
+
+To format each node, we use the `getNodeFormatter` method of the `AbstractFormatter`. The resulting generic `NodeFormatter<T extends AstNode>` provides us with methods to select specific parts of a parsed `AstNode` such as properties or keywords.
+
+Once we have selected the nodes of our document that we are interested in formatting, we can start applying a specific formatting.
+Each formatting option allows to prepend/append whitespace to each note.
+The `Formatting` namespace provides a few predefined formatting options which we can use for this.
+ 
+We first start off by formatting the `Domainmodel` element of our DSL.
+It is the root node of every document and just contains a list of other elements.
+These elements need to be realigned to the root of the document in case they are indented.
+We will use the `Formatting.noIndent` options for that:
+
+```ts
+if (ast.isDomainmodel(node)) {
+    // Create a new node formatter
+    const formatter = this.getNodeFormatter(node);
+    // Select a formatting region which contains all children
+    const nodes = formatter.nodes(...node.elements);
+    // Prepend all these nodes with no indent
+    nodes.prepend(Formatting.noIndent());
+}
+```
+
+Our other elements, namely `Entity` and `PackageDeclaration`, can be arbitrarily deeply nested, so using `noIndent` is out of the question for them.
+Instead we will use `indent` on everything between the `{` and `}` tokens. The formatter internally keeps track of the current indentation level:
+
+```ts
+if (ast.isEntity(node) || isPackageDeclaration(node)) {
```

Comment:
This code differs with what we have for the domain model in the repo [here](https://github.com/langium/langium/blob/main/examples/domainmodel/src/language-server/domain-model-formatter.ts). Maybe update repo's version?

Reply:
We should do that, but it's unrelated to this PR.